scripts/shared/cloud-prepare.sh

```diff
@@ -27,18 +27,55 @@ function check_gateway_exists() {
     [[ $(kubectl get nodes -l "${GATEWAY_LABEL}" --no-headers | wc -l) -gt 0 ]]
 }
 
-function prepare_kind() {
-    read -r -a nodes <<< "${cluster_nodes[$cluster]}"
-    local node=${cluster}-${nodes[-1]}
-    kubectl label node "$node" "$GATEWAY_LABEL" --overwrite
```

Comment:
Why remove the labeling from here?
Now this call is duplicated 3 times..

Reply:
You could just assemble a list of nodes here and label all of them in a loop..

Also labeling still needs to happen per node just like air gap handling

Reply:
Fixed.